### ISSUE TITLE
Can't run or debug junit5 'TestTemplate' and 'Meta Annotation' case

### ISSUE THREAD

**Issue by ZhengJun003** [microsoft/vscode-java-test#859]
### Environment:
OS: Windows
JDK: 8/12
Java Extensions:

- Language Support for Java(TM) by Red Hat： 0.52.0
- Debugger for Java: 0.23.0
- Java Test Runner: 0.21.0

### Steps to reproduce:

1. Open the JUnit5 project in VSCode: https://github.com/ZhengJun003/junit5-samples/tree/testrunner
2. Run `MyFastTestIT `and `TestTemplateI `in the test explorer

### Result:
![image](https://user-images.githubusercontent.com/41051257/67749020-30dee400-fa67-11e9-9377-a0a5939f2ef8.png)

![image](https://user-images.githubusercontent.com/41051257/67749268-c37f8300-fa67-11e9-99c5-5f3461b0ef82.png)

**Comment by ZhengJun003** [comment]
Java file's name and class name must be the same, otherwise, maven will skip these cases

**Comment by jdneo** [comment]
So it turns out that this is a Eclipse side bug. So I'll reopen it until we fix it in the upstream.

**Comment by geist-2501** [comment]
Hi, I'm having a similar issue. When I try to run tests in my project, I receive the same 'Error: Failed to parse the JUnit launch arguments'. It's for a University project, so I can't use Maven or Gradle, so my project set-up is the '[invisible](https://code.visualstudio.com/docs/java/java-project#_standalone-java-file-support)' type.

What's strange is that I had a similar project of the same layout that I managed to run tests on, which now no longer works, and gives the same error.

Github link -> [https://github.com/geist-2501/CS2001-W05-StacksQueues](https://github.com/geist-2501/CS2001-W05-StacksQueues)

### Environment;

- Debugger for Java 0.23.0
- Java Test Runner 0.21.0
- Language Support 0.52.0

### How to repro;

1. Open project in VSCode.
2. Open test runner and attempt to run.

I have a feeling I'm missing something really stupid, I hope you can help!

**Comment by ZhengJun003** [comment]
@geist-2501, please download the "junit-platform-console-standalone-1.5.2" jar in your project's lib folder.

![image](https://user-images.githubusercontent.com/41051257/68096048-3914a480-fee9-11e9-8441-9294f28aef84.png)

**Comment by geist-2501** [comment]
@ZhengJun003 Thanks for getting back!
I've downloaded the same `junit-platform-console-standalone-1.5.2.jar`, which has got it closer to working, but now I see a different error message upon runnnig a test:
![ErrorMessage](https://user-images.githubusercontent.com/15649238/68221133-e0c0ce00-ffe0-11e9-85c6-6248b4e5a259.png)
What changed from 0.20.0 to 0.21.0 that means it now requires the `standalone`?

**Comment by jdneo** [comment]
@geist-2501 This seems to a VS Code bug. According to: https://github.com/microsoft/vscode-go/issues/2831#issuecomment-544271725

Would you mind to try it in the latest VS Code Insider?

The changes we made from `0.20.0` to `0.21.0` is to use the Eclipse's JUnit runner instead of our own runners. So the jars in the classpath has been changed.

**Comment by geist-2501** [comment]
@jdneo Yep, just tried it with the latest insiders version and it worked! Thanks!

**Comment by OneBigOwnage** [comment]
It took me a while to find this issue, as I had this problem without using a "secondary class". Downgraded to 20.0 and it works perfectly again.

When will this be fixed in the non-insiders version of VSCode?

**Comment by jdneo** [comment]
@OneBigOwnage VS Code team released `1.40.0` recently. Is this problem still occur in `1.40.0` on your side?

**Comment by rajkundu** [comment]
> Hi, I'm having a similar issue. When I try to run tests in my project, I receive the same 'Error: Failed to parse the JUnit launch arguments'. It's for a University project, so I can't use Maven or Gradle, so my project set-up is the '[invisible](https://code.visualstudio.com/docs/java/java-project#_standalone-java-file-support)' type.
> 
> What's strange is that I had a similar project of the same layout that I managed to run tests on (Oct 14th), which now no longer works, and gives the same error.
> 
> Github link -> [here](https://github.com/geist-2501/JavaTestExample)
> ### Environment;
> 
>     * Debugger for Java 0.23.0
> 
>     * Java Test Runner 0.21.0
> 
>     * Language Support 0.52.0
> 
> 
> ### How to repro;
> 
>     1. Open project in VSCode.
> 
>     2. Open test runner and attempt to run.
> 
> 
> I have a feeling I'm missing something really stupid, I hope you can help!
> 
> Update (3/11/19): Rolling back to Java Test Runner 0.20.0 fixed the issue! It must be something with the new version.

I had the same exact issue as you. I am working on a university project, too, and had the following error when running JUnit tests: "Error: Failed to parse the JUnit launch arguments." (VS Code v1.40.0, JUnit Test Runner v0.21.0)

So, I tried your suggestion of rolling back the extension, and JUnit tests run perfectly on VS Code v1.40.0 with JUnit Test Runner v0.20.0. Thank you!

**Comment by jdneo** [comment]
@rajkundu Is it working on VS Code 1.40.0 with Test Runner 0.21.0?

**Comment by geist-2501** [comment]
@jdneo @rajkundu Using VS Code 1.40.0 and Test Runner 0.21.0 works for me now, as long as I have the [Junit Standalone Console](https://search.maven.org/artifact/org.junit.platform/junit-platform-console-standalone/1.5.2/jar) .jar in a `lib` folder

**Comment by jdneo** [comment]
Ok so seems the problem has been solved. So I'm closing this issue.

Here is the final solution:
1. Update VS Code to 1.40.0
2. Update Test Runner to 0.21.0
3. Have the [Junit Standalone Console](https://search.maven.org/artifact/org.junit.platform/junit-platform-console-standalone/1.5.2/jar) jar in a `lib` folder

**Comment by vwxyzjn** [comment]
@jdneo Thanks that works!

**Comment by moinessim** [comment]
Had the same problem, read [this](https://gitmemory.com/issue/microsoft/vscode-java-test/859/552281046) rolled back to v0.20.0 and it worked.

**Comment by jdneo** [comment]
@moinessim Have you tried [this](https://github.com/microsoft/vscode-java-test/issues/859#issuecomment-552433039) solution?

**Comment by moinessim** [comment]
> @moinessim Have you tried [this](https://github.com/microsoft/vscode-java-test/issues/859#issuecomment-552433039) solution?

Hi @jdneo, I tried the following, but had the same error message. 
Adding the Junit Standalone Console jar: 
- in a lib folder in the project root directory.
- in a lib folder next to the test class.
- in the pom.xml as a Maven dependency with test scope.
